packages/notifications/src/middleware.ts

```diff
@@ -0,0 +1,70 @@
+import { Action, Dispatch, MiddlewareAPI } from 'redux';
+import {
+  TNotification,
+  TNotificationOnDismiss,
+  TNotificationAction,
+} from './types';
+import { ADD_NOTIFICATION, REMOVE_NOTIFICATION } from './action-types';
+import { removeNotification } from './action-creators';
+
+// Force TS cast of generic action to TNotificationAction
+const isNotificationAction = <Payload extends TNotification>(
+  action: Action
+): action is TNotificationAction<Payload> =>
```

Comment:
oh 🤩

Reply:
It is a type predicate which is part of the whole type guards thing. It is a bit like an ‚as‘ but safer and used in return types.

Reply:
Yeah, Philipp actually gave me the hint 😉